lib/discovery/index.js

```diff
@@ -234,6 +235,9 @@ export class Discovery extends TypedEmitter {
    * @returns {Promise<void>}
    */
   async _onMdnsConnection(socket) {
+    // end connection if its from a non-local node
+    if(!bogon(socket.remoteAddress)) socket.end()
```

Comment:
```suggestion
    if(!bogon.isPrivate(socket.remoteAddress)) socket.end()
```
We're concerned about checking this is from a local IP, not any Bogon.

Reply:
Ooops, yeah, fixed that. 
Additionally, `socket.remoteAddress` returns an IPv6 with the IPv4 emebed (is it always like that or does it depend on some flag?); so I need to strip the `:ffff::` to avoid false positives on the check